ibc-core/ics04-channel/types/src/acknowledgement.rs

```diff
@@ -142,3 +142,9 @@ impl From<AcknowledgementStatus> for Acknowledgement {
             .expect("token transfer internal error: ack is never supposed to be empty")
     }
 }
+
+impl From<DecodingError> for StatusValue {
+    fn from(e: DecodingError) -> Self {
+        StatusValue::new(e.to_string()).expect("error message must not be empty")
+    }
+}
```

Comment:
It looks this is introduced inadvertantly.

Reply:
It's actually needed in a few places, such as [here](https://github.com/cosmos/ibc-rs/blob/imp/define-host-error/ibc-apps/ics721-nft-transfer/src/module.rs#L176-L180)

Reply:
I wonder, didn’t we revert the `AcknowledgementStatus` error in the previous PR to how it was before, instead of using `DecodingError` in such places? to avoid causing a state-machine breaking change?

Reply:
You're right; I forgot about that.

Reply:
[1912395](https://github.com/cosmos/ibc-rs/pull/1331/commits/191239572cda876fccfba0f072ba062f2c6baafa)